server/src/test/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintMonitorTests.java

```diff
@@ -0,0 +1,472 @@
+            testState.mockRerouteService
+        );
+
+        try (MockLog mockLog = MockLog.capture(WriteLoadConstraintMonitor.class)) {
+            mockLog.addExpectation(
+                new MockLog.SeenEventExpectation(
+                    "don't reroute due to all nodes exceeding utilization threshold",
+                    WriteLoadConstraintMonitor.class.getCanonicalName(),
+                    Level.DEBUG,
+                    "No nodes are suitable as relocation targets"
+                )
+            );
+            writeLoadConstraintMonitor.onNewInfo(
+                ClusterInfo.builder().nodeUsageStatsForThreadPools(nodeUsageStatsWithHighUtilization).build()
+            );
+            mockLog.assertAllExpectationsMatched();
+            verifyNoInteractions(testState.mockRerouteService);
+        }
+    }
+
+    @TestLogging(
+        value = "org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitor:DEBUG",
+        reason = "ensure we're skipping reroute for the right reason"
+    )
+    public void testRerouteIsNotCalledWhenNoNodeIsUnderLatencyThreshold() {
+        final TestState testState = createRandomTestStateThatWillTriggerReroute();
+
+        final ClusterInfo clusterInfoWithAllNodesOverLatencyThreshold = createClusterInfoWithHotSpots(
+            testState.clusterState,
+            testState.numberOfNodes,
+            testState.latencyThresholdMillis,
+            testState.highUtilizationThresholdPercent
+        );
+
+        final WriteLoadConstraintMonitor writeLoadConstraintMonitor = new WriteLoadConstraintMonitor(
+            testState.clusterSettings,
+            testState.currentTimeSupplier,
+            () -> testState.clusterState,
+            testState.mockRerouteService
+        );
+
+        try (MockLog mockLog = MockLog.capture(WriteLoadConstraintMonitor.class)) {
+            mockLog.addExpectation(
+                new MockLog.SeenEventExpectation(
+                    "don't reroute due to all nodes exceeding latency threshold",
+                    WriteLoadConstraintMonitor.class.getCanonicalName(),
+                    Level.DEBUG,
+                    "No nodes are suitable as relocation targets"
+                )
+            );
+            writeLoadConstraintMonitor.onNewInfo(clusterInfoWithAllNodesOverLatencyThreshold);
+            mockLog.assertAllExpectationsMatched();
+            verifyNoInteractions(testState.mockRerouteService);
+        }
+    }
+
+    @TestLogging(
+        value = "org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitor:DEBUG",
+        reason = "ensure we're skipping reroute for the right reason"
+    )
+    public void testRerouteIsNotCalledAgainBeforeMinimumIntervalHasPassed() {
+        final TestState testState = createRandomTestStateThatWillTriggerReroute();
+        final TimeValue minimumInterval = testState.clusterSettings.get(
+            WriteLoadConstraintSettings.WRITE_LOAD_DECIDER_REROUTE_INTERVAL_SETTING
```

Comment:
I expect this test will need a non default INTERVAL setting to be meaningful now.

Reply:
Done in 2abd946b8a2b78f2d923355bad371fba80bc8f8c